### DISCUSSION TITLE
Importing ApiClient from telegraf module

### DISCUSSION THREAD

**Discussion by amirfarzamnia** [telegraf/telegraf#1933] [Q&A] [answered]
The Telegram class has been expanded to include features from the ApiClient class, which is found in the 'telegraf/src/core/network/client' location. The ApiClient class comes with extra customization options for better accessibility. To make integration smoother, I suggest updating the telegraf package to allow importing the ApiClient like this:

```javascript
const { Telegraf, ApiClient } = require('telegraf');
```

**Comment by amirfarzamnia** [comment]
This can be really useful in my case; probably it can help develop bots with less coding.

**Comment by MKRhere** [comment]
> extra customization options

Which customisation options are you missing from the `Telegraf` class? All `ApiClient` options can also be passed to the main Telegraf class:

```TS
const bot = new Telegraf(token, { telegram: clientOpts });
```

Also, if you prefer, you can import the `Telegram` class, which extends `ApiClient`.

```TS
const { Telegram } = require("telegraf");
const telegram = new Telegram(token, clientOpts);
```

**Reply by amirfarzamnia** [reply]
In my case, utilizing the ApiClient class proves to be a more efficient approach, as it requires only the following code snippet:

```javascript
const client = new ApiClient(token);

client.callApi('sendMessage', {
   // Any required parameters
});
```

This approach facilitates greater customization, exemplified here:

```javascript
// Importing ApiClient from the 'telegraf' module.
const { ApiClient } = require('telegraf');

// Creating a new Api client for request management.
const client = new ApiClient('token-123456');

// Array of requests to be sent.
const reqEndpoints = [
	{
		endpoint: 'sendMessage',
		param: {
			text: 'John Doe',
			chat_id: '123456'
		}
	},
	{
		endpoint: 'setChatTitle',
		param: {
			phone_number: 'John Doe Developers',
			chat_id: '123456'
		}
	}
];

// Invoking the client for each request to send them.
reqEndpoints.forEach(({ endpoint, param }) => client.callApi(endpoint, param));
```

This approach allows for the seamless dispatch of as many requests as necessary with minimal code. Moreover, it enhances customization accessibility by distinctly separating the endpoint on the left side and the parameters on the right side.

Presently, we are obliged to employ the following code for requiring the ApiClient, which is not recommended in programming practices:

```javascript
const { default: ApiClient } = require('./node_modules/telegraf/lib/core/network/client.js');
```

**Reply by amirfarzamnia** [reply]
It's also worth mentioning that when a chatbot needs to handle more than 50 endpoints for sending requests, utilizing the current syntax would necessitate approximately 200 lines of code, potentially leading to user errors. However, by employing the ApiClient, this can be streamlined to around 40 lines of code, significantly enhancing the ease and speed of debugging.

**Reply by MKRhere** [reply, marked as answer]
The Telegram class is just `ApiClient` with more methods:

```TS
const { Telegram } = require("telegraf");
const telegram = new Telegram(token);
telegram.callApi(method, opts);
```

If that's all you need, that's the code you need.

**Reply by amirfarzamnia** [reply]
Thanks
